openmls/src/treesync/diff.rs

```diff
@@ -83,59 +92,61 @@ impl<'a> From<&'a TreeSync> for TreeSyncDiff<'a> {
 impl<'a> TreeSyncDiff<'a> {
     /// Check if the right-most leaf and its parent are blank. If that is the
     /// case, remove the right-most leaf and its parent until either the
-    /// right-most leaf or its parent are not blank anymore. This behaviour
-    /// differs from that in the MLS spec. This is because there is a suspected
-    /// bug in the trimming algorithm in the spec, which is currently under
-    /// discussion.
-    pub(crate) fn trim_tree(&mut self) -> Result<(), TreeSyncDiffError> {
+    /// right-most leaf or its parent are not blank anymore.
+    pub(crate) fn trim_tree(&mut self) {
         // Nothing to trim if there's only one leaf left.
         if self.leaf_count() == 1 {
-            return Ok(());
+            return;
         }
-        let mut leaf_index = self.diff.leaf(self.leaf_count() - 1);
-        let mut parent_index = self.diff.parent(leaf_index)?;
-        // Trim only if the parent node is blank as well;.
-        while self.diff.node(leaf_index)?.node().is_none()
-            && self.diff.node(parent_index)?.node().is_none()
-        {
-            self.diff.remove_leaf()?;
-            // If there's only one leaf left, it won't have a parent, so we'll
-            // have to stop here.
-            if self.leaf_count() == 1 {
-                return Ok(());
+
+        // We reverse the nodes here to make sure we start with the right-most
+        let mut nodes = self
+            .diff
+            .nodes()
+            .collect::<Vec<(TreeNodeIndex, &TreeSyncNode)>>();
+        nodes.reverse();
```

Comment:
Since everything is done via an iterator now, have you considered implementing `DoubleEndedIterator` for `SortedIter` and just using `rev()` here? Not sure if it's worth the effort, but it would probably fit the theme better than collecting and looping over `chunks`.